src/controllers/changeset.controllers.js

```diff
@@ -132,28 +132,25 @@ export const createOrUpdateChangesetFile = async (payload) => {
     );
   } catch (error) {
     const changesetFilePath = `${CHANGESET_PATH}/${prNumber}.yml`;
-    // Delete changeset file if one was previously created
+    // Get changeset file if one exists
     const changesetFile = await getFileByPath(
       headOctokit,
       headOwner,
       headRepo,
       headBranch,
       changesetFilePath
     );
-    if(changesetFile) {
-      const commitMessage = `Changeset file for PR #${prNumber} deleted`;
-      await deleteFileByPath(
-        headOctokit,
-        headOwner,
-        headRepo,
-        headBranch,
-        changesetFilePath,
-        commitMessage,
-        changesetFile?.sha
-      );
-    } else {
-      console.log("No changeset file to delete.");
-    }
```

Comment:
I changed the logic in order to avoid an unnecessary API call. As it was, we checked to see if the file existed, and, whether or not it existed, we still called the `deleteFile` API endpoint. By adding an `if` statement here, we don't need to rely on the `deleteFile` API call returning with a 404 error .

Reply:
Can you move the logic inside the deleteFileByPath() function?

Reply:
Good idea. I just created an issue for this in our repository.